docs/source/data/kedro_dvc_versioning.md

```diff
@@ -0,0 +1,187 @@
+
+```bash
+git checkout <commit hash> data/01_raw/companies.csv.dvc
+dvc checkout
+dvc pull
+```
+
+## Versioning with DVC Data Pipelines
+
+While the previous method allows you to version datasets, it comes with some limitations:
+
+- Intermediate and output datasets must be added to DVC manually.
+- Parameters and code changes are not explicitly tracked.
+- Artefacts and metrics cannot be tracked effectively.
+
+To address these issues, you can define Kedro pipelines as DVC stages in the dvc.yaml file.
+
+### Defining Kedro Pipelines as DVC Stages
+
+Here is an example configuration for dvc.yaml:
+
+```yaml
+stages:
+  data_processing:
+    cmd: kedro run --pipeline data_processing
+    deps:
+      - data/01_raw/companies.csv
+      - data/01_raw/reviews.csv
+      - data/01_raw/shuttles.xlsx
+    outs:
+      - data/02_intermediate/preprocessed_companies.pq
+      - data/02_intermediate/preprocessed_shuttles.pq
+      - data/03_primary/model_input_table.pq
+
+  data_science:
+    cmd: kedro run --pipeline data_science
+    deps:
+      - data/03_primary/model_input_table.pq
+    outs:
+      - data/06_models/regressor.pickle
+```
+
+Run the pipeline with:
+
+```bash
+dvc repro
+```
+
+### Updating a Dataset
+
+If one of the datasets is updated, you can rerun only the pipelines affected by the change.
+
+The command `dvc repro` executes pipelines where outputs or dependencies have changed.
+
+### Tracking Code Changes
+
+You can track changes to your code by adding the relevant files to the `deps` section in `dvc.yaml`.
+
+```yaml
+stages:
+  data_processing:
+    cmd: kedro run --pipeline data_processing
+    deps:
+      - data/01_raw/companies.csv
+      - data/01_raw/reviews.csv
+      - data/01_raw/shuttles.xlsx
+      - src/space_dvc/pipelines/data_processing/nodes.py
+      - src/space_dvc/pipelines/data_processing/pipeline.py
+    outs:
+      - data/02_intermediate/preprocessed_companies.pq
+      - data/02_intermediate/preprocessed_shuttles.pq
+      - data/03_primary/model_input_table.pq
+```
+
+After applying the desired code changes, run:
+
+```bash
+dvc repro
+dvc push
+```
+
+### Tracking Parameters
+
+To track parameters, you can include them under the params section in `dvc.yaml`.
+
+```yaml
+stages:
+  data_science:
+    cmd: kedro run --pipeline data_science
+    deps:
+      - data/03_primary/model_input_table.pq
+      - src/space_dvc/pipelines/data_science/nodes.py
+      - src/space_dvc/pipelines/data_science/pipeline.py
+    params:
+      - conf/base/parameters_data_science.yaml:
+          - model_options
+    outs:
+      - data/06_models/regressor.pickle
+```
+
+Run the pipeline and push the changes:
+
+```bash
+dvc repro
+dvc push
+```
+
+### Running Experiments with Different Parameters
+
+To experiment with different parameter values, update the parameter in `parameters.yaml` and then run the pipelines with `dvc repro`.
+
+Compare parameter changes between runs with `dvc params diff`
```

Comment:
Overall looks good so far! Would be nice to also add outputs to some of these steps.